pallets/moonbeam-xcm-benchmarks/src/weights/mod.rs

```diff
@@ -27,31 +27,32 @@ use xcm::{
 	DoubleEncoded,
 };
 
+const MAX_ASSETS: u32 = 100;
+
 trait WeighMultiAssets {
 	fn weigh_multi_assets(&self, weight: Weight) -> XCMWeight;
 }
 
 trait WeighMultiAssetsFilter {
-	fn weigh_multi_assets_filter(&self, max_assests: u32, weight: Weight) -> XCMWeight;
+	fn weigh_multi_assets_filter(&self, weight: Weight) -> XCMWeight;
 }
 
 impl WeighMultiAssetsFilter for MultiAssetFilter {
-	fn weigh_multi_assets_filter(&self, max_assests: u32, weight: Weight) -> XCMWeight {
-		let weight = match self {
+	fn weigh_multi_assets_filter(&self, weight: Weight) -> XCMWeight {
+		match self {
 			Self::Definite(assets) => {
 				weight.saturating_mul(assets.inner().into_iter().count() as u64)
 			}
-			Self::Wild(_) => weight.saturating_mul(max_assests as u64),
-		};
-		weight.ref_time()
+			Self::Wild(AllOf { .. } | AllOfCounted { .. }) => weight,
+			Self::Wild(AllCounted(count)) => weight.saturating_mul(*count as u64),
+			Self::Wild(All) => weight.saturating_mul(MAX_ASSETS as u64),
```

Comment:
Good! I think this will not disrupt anything. V2-V3 conversions will be handled with the counted variant, and we already modified xtokens to use the counted variant in V3.